persistent/picklecache.py

```diff
@@ -106,32 +187,26 @@ def get(self, oid, default=None):
     def mru(self, oid):
         """ See IPickleCache.
         """
-        node = self.ring.next
-        while node is not self.ring and node.object._p_oid != oid:
-            node = node.next
-        if node is self.ring:
-            value = self.data[oid]
-            if value._p_state != GHOST:
```

Comment:
Did you measure costs of out-lining all the ring code?  Extra function calls ain't cheap.